Close some sources of fake tensor leakages

tugsbayasgalan opened the pull request:
Differential Revision: D79694055

Couple of fixes:
1. When we run into an operation we didn't proxy, we end up emitting fake constants. We detect this and error using the FQN of the lifted constant 
2. Previous attribute mutation detection logic in non-strict didn't account for nested module structure. This fixes silent incorrectness issue of exporting esm and qwen in non-strict
3. We modify yolov3 to fix the previous silent incorrect behaviour 

When upgrading torchbench pin, opacus_cifar10 seems to not run on eager anymore. I verified this by pushing a temporary PR on master with new pin. So i added it to expect_fail list. 






cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

avikchaudhuri reviewed (approved):
Nice

zhxchen17 reviewed (dismissed):
We should verify this PR works on https://github.com/pytorch/pytorch/issues/160066 before landing

tugsbayasgalan committed:
Fix tensor leak when nested in non-strict (#159923)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/159923

Test Plan:
CI

Rollback Plan:

Reviewed By: muchulee8

Differential Revision: D79694055

zou3519 commented:
@tugsbayasgalan this PR is breaking Inductor periodic CI. Since I don't think we'll have a fix soon (it's Friday), I'm going to revert unless you think you can forward-fix this today.

zou3519 commented:
@pytorchbot help

jeanschmidt commented:
@pytorchbot merge -f "Breaking ship-it"
